### PR TITLE
Feat/grd cli cog

### PR DESCRIPTION
**Pull request by wildintellect**
Before you submit a pull request, please fill in the following:

**Related Issue(s):**
This is related to adding GRD Collection

**Description:**
- Adds archive format option to cli for testing SAFE vs COG
- Adds a new example in the data-files of a different product, EW with DH (HH+HV) polarizations, and the COG instead of SAFE format.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [ ] Documentation has been updated to reflect changes, if applicable.
- [ ] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: src/stactools/sentinel1/grd/commands.py [hunk 1]
```diff
@@ -34,14 +37,24 @@ def create_collection_command(destination: str) -> None:
 )
 @click.argument("src")
 @click.argument("dst")
-def create_item_command(src: str, dst: str) -> None:
+@click.option("--format", default="SAFE", type=str, help="SAFE or COG format")
+def create_item_command(
+    src: str, dst: str, href: Optional[str] = None, format: str = "SAFE"
```
Comment (jjfrench) [inline src/stactools/sentinel1/grd/commands.py]: is href just always None / unused?

**Comment by wildintellect** [comment]
TODO: needs a new test of the cli specific to COG.

**Review by jjfrench — APPROVED** [review]

File: src/stactools/sentinel1/grd/commands.py (hunk 1 above)
Comment (wildintellect) [reply src/stactools/sentinel1/grd/commands.py]: I meant to leave it out of this PR, it's a more complicated argument that isn't critical to the COG fix.

**Review by jjfrench — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**